src/main/java/Parser.java

```diff
@@ -0,0 +1,93 @@
+import java.util.Scanner;
+/**
+ * Represents a parser of user commands in Duke chat bot.
+ */
+public class Parser {
+    protected Scanner sc;
+
+    public Parser(Scanner sc) {
+        this.sc = sc;
+    }
+
+    /**
+     * Reads and react to user command
+     * @param tasks Current task list.
+     */
+    public void read(TaskList tasks) {
+        while(true) {
+            String cmd = sc.nextLine();
+            Scanner cmdSc = new Scanner(cmd);
+            if (cmd.toLowerCase().equals("bye")) {
+                System.out.println("Bye. Hope to see you again soon!");
+                break;
+            } else {
+                String cmdWord = cmdSc.next().toLowerCase();
+                switch (cmdWord) {
+                    case "todo":
+                        try {
+                            if (cmdSc.hasNext()) {
+                                tasks.addToDo(cmdSc.nextLine());
+                            } else {
+                                throw new DukeException("The description of a todo cannot be empty.");
+                            }
+                        }
+                        catch (DukeException exp) {
+                            System.out.println("OOPS!!! " + exp.getMessage());
```

Comment:
Same issue, I think potentially you can refactor such that the UI handles all logic of what to print?